src/tests/run_tests.sh

```diff
@@ -2,18 +2,32 @@
 
 three_tests=( "cachestat" "dc" "fd" "mount" "process" "shm" "socket" "swap" "sync" "vfs" )
 one_test=( "disk" "hardirq" "oomkill" "softirq" )
+end_loop=
 
 echo "Running all tests with three options"
 for i in "${three_tests[@]}" ; do
     {
-        echo "================  Running $i  ================"
-        echo "---> Probe: "
-        "./$i" --probe
-        echo "---> Tracepoint: "
-        "./$i" --tracepoint
-        echo "---> Trampoline: "
-        "./$i" --trampoline
-        echo "  "
+        pid=$("./$i" --help | grep pid)
+        if [ -z "${pid}" ]; then
+    	    end_loop=0
+        else
+    	    end_loop=2
+        fi
+
+        for j in $(seq 0 $end_loop); do
+	    if [ -z "${pid}" ]; then
+                printf "================  Running %s without PID  ================" "${i}"
+	    else
+                printf "================  Running %s with PID GROUP %s  ================" "${i}" "${j}"
```

Comment:
Could you elaborate a bit on what is happening here? As I see it, the test just runs three times in a row - you don't use `j` anywhere. What's the purpose of that?

Reply:
Of course, let me explain the idea. Users can monitor PID in different ways (`real_parent`, `parent`, all PID), and this is only possible for some of process inside `three_tests`. 

This pair of `if .. else` was added to create a proper header for the tests, that I expect to bring with PR https://github.com/netdata/ebpf-co-re/pull/25. I am identifying the `PID GROUP`  according to its id.